src/shared/install-file.h

```diff
@@ -4,11 +4,13 @@
 #include "shared-forward.h"
 
 typedef enum InstallFileFlags {
-        INSTALL_REPLACE    = 1 << 0, /* Replace an existing inode */
-        INSTALL_READ_ONLY  = 1 << 1, /* Call fs_make_very_read_only() to make the inode comprehensively read-only */
-        INSTALL_FSYNC      = 1 << 2, /* fsync() file contents before moving file in */
-        INSTALL_FSYNC_FULL = 1 << 3, /* like INSTALL_FSYNC, but also fsync() parent dir before+after moving file in */
-        INSTALL_SYNCFS     = 1 << 4, /* syncfs() before moving file in, fsync() parent dir after moving file in */
+        INSTALL_REPLACE            = 1 << 0, /* Replace an existing inode */
+        INSTALL_READ_ONLY          = 1 << 1, /* Call fs_make_very_read_only() to make the inode comprehensively read-only */
+        INSTALL_READ_ONLY_GRACEFUL = 1 << 2, /* As above, but don't fail if this doesn't work */
```

Comment:
Do we need two separate flags or would `INSTALl_GRACEFUL` do the trick as well?

Reply:
Some unit tests for the new flags would be very nice given the install_file() function is becoming quite complex with all these flags.